MidCamp/midcamp
# Workflow... shmerkflow

# Question

I set up a lighter alternative approach over at #324 .  It seems that once we set the Topic node's default publishing status to "unpublished" we solve a lot of problems.  See https://www.drupal.org/project/drupal/issues/2886569.  Users without "administer content" permissions don't see the "Publishing Status" checkbox on the node add/edit form anyway, so we can bypass all the overhead here.

This will help us avoid the different routes for form modes like `node/[NID]/edit/[FORM-MODE]` and keep things a bit simpler.

## To Test

https://nginx-midcamp-org-feature-workflow-shmerkflow.us.amazee.io/

- Log in as an authenticated user
- Create a new Topic proposal at `/node/add/topic/limited`.  Confirm you can save but not publish.

## Description

This PR does the following:

- Adds update hook to clean up moderation states from the database
- Removes workflow-related form_alters
- Uninstalls Content Moderation and Workflows
- Adds/installs [Form Mode Manager](https://www.drupal.org/project/form_mode_manager) 
- Creates new entity form displays `admin` and `limited`
- Sets `field_topic_type` to optional on Topic nodes
- Sets `field_topic_type` to private
- Sets `authenticated` user permissions to use the new `limited` form  mode
- Sets `content_editor` and `organizer` user permissions to use the new `admin` form  mode
- Defaults Topic nodes to unpublished, ensures `limited` form does not have the field widget to change this.
